### DISCUSSION TITLE
rabies_preprocess.pkl missing

### DISCUSSION THREAD

**Discussion by HelloKitty4lif3** [CoBrALab/RABIES#358] [Q&A] [answered]
Hi

I'm trying to run confound correction but receive a "FileNotFoundError: [Errno 2] No such file or directory: 'preprocess_outputs/rabies_preprocess.pkl'" message. Should I have this file after running preprocessing ?  I Run this as my preprocessing.
![Screenshot 2024-02-29 142405](https://github.com/CoBrALab/RABIES/assets/161685735/ceecbe97-c444-4e36-86a2-930c626b03b2)

**Comment by Gab-D-G** [comment]
Hi, yes this file should be there right when you execute the command. Either nothing was executed, or you deleted the file somehow. What gets printed in your terminal?

**Reply by colon-perez** [reply]
I actually was coming in to describe the same problem but using docker. I started a thread at docker to see if its an issue with the container but it doesn't seem to. The preprocess directory has the file and its not empty, I checked within the container and it seems to be loaded with read permissions so I am stuck as to why this is happening. 

I am using the latest version of rabies this one "ghcr.io/cobralab/rabies:latest" 

Another note, I am also using SIGMA. So I don't know if there is a bug when mounting alternate atlases to rabies.

**Reply by colon-perez** [reply]
The issue is in the path setting (at least for me). Make sure you set absolute paths (slash before and after the directory, statements below between asterisks), and it should work. Note mine is set for docker, so I mount the directory onto the container, so the absolute path in docker and locally (as you are working) may be different but I hope this helps.

`rabies -p MultiProc --local_threads 5 confound_correction **/preprocess_outputs/ /confound_correction_outputs/** ....`

**Reply by colon-perez** [reply]
Thanks, Gabriel, for being so responsive on this discussion board.

**Reply by Gab-D-G** [reply]
Indeed, absolute paths should always be provided when using Docker/Singularity containers. Thank you for the input @colon-perez . If using a local installation, as seems to be the case with @HelloKitty4lif3 , then it can be fine to use relative paths. Docker can indeed lead to permission issues, if you are still encountering them, consider looking up the examples with `--user $(id -u)` shown here https://rabies.readthedocs.io/en/latest/running_the_software.html#:~:text=%2D%2Duser%20%24(id%20%2Du)%20can%20be%20added%20to%20mitigate%20writing%20permission%20issues%20when%20using%20Docker.

**Reply by colon-perez** [reply, marked as answer]
Mine is working now. I mentioned the path because working directories may have been altered as @HelloKitty4lif3 was working on her dataset and now the relative path may wrong. Hopefully they figure it out.

**Reply by HelloKitty4lif3** [reply]
Thank you for the help @colon-perez and @Gab-D-G. It was a typo of the preprocess path. However, I am met with the same error now but with a missing rabies_preprocess_workflow.pkl now. 

<img width="283" alt="image" src="https://github.com/CoBrALab/RABIES/assets/161685735/21f95c5b-f155-4410-8274-b08a16bdbdd8">

Any suggestions where I might have gone wrong?

**Reply by Gab-D-G** [reply]
This means the preprocessing was not completed. Can you inspect the log for an error?

**Reply by gdevenyi** [reply]
User is not using the container, they need to reproduce this using the supported release.
